### ISSUE TITLE
Does match_bucket_filter support query_string filters?

### ISSUE THREAD

**Issue by oridayne** [Yelp/elastalert#1099]
I get the error message no [query] registered for [query] if I try to use a query_string

**Comment by Qmando** [comment]
Thanks for the report. There is some special code that handles this for filter but not for bucket filter. You can work around this by removing the top level "query"

```
match_bucket_filter:
 - query_string:
      query: "blah blah"
```

**Comment by mircopolo** [comment]
I don't think it is quite the same issue mind, the dsl for the filter aggregation appears unchanged going back to 1.x. 

The query{} element is not actually a valid query clause rather it signifies search context, and not sure it was ever valid as a child element of a filter{} context. 

Only leaf query clauses like match, term, query_string or compound query clauses such a bool can go directly under the filter element.

I should probably have made the docs a bit clearer  that it is not exactly the same as the main search filter.

**Comment by Qmando** [comment]
True. I got used to starting things with query and then popping that when doing aggregations, like https://github.com/Yelp/elastalert/blob/master/elastalert/elastalert.py#L220 and  https://github.com/Yelp/elastalert/blob/master/elastalert/elastalert.py#L883 instead of encouraging people to avoid starting with query.
